docs/faq.rst

```diff
@@ -65,6 +65,136 @@ Additional reading:
+
+  jax.jit(lambda arr: np.split(arr, 2, 0))(np.zeros(4))
+
+**Note a new closure is created at every invocation, which defeats the
+compilation caching mechanism, which is why static_argnums is preferred.**
+
+To understand more subtleties having to do with tracers vs. regular values, and
+concrete vs. abstract values, you may want to read `Different kinds of JAX values`_.
+
+Different kinds of JAX values
+------------------------------
+
+In the process of transforming functions, JAX replaces some some function
+arguments with special tracer values.
+You could see this if you use a ``print`` statement::
+
+  def func(x):
+    print(x)
+    return np.cos(x)
+
+  res = jax.jit(func)(0.)
+
+The above code does return the correct value ``1.`` but it also prints
+``Traced<ShapedArray(float32[])>`` for the value of ``x``. Normally, JAX
+handles these tracer values internally in a transparent way, e.g.,
+in the numeric JAX primitives that are used to implement the
+``jax.numpy`` functions. This is why ``np.cos`` works in the example above.
+
+More precisely, a **tracer** value is introduced for the argument of
+a JAX-transformed function, except the arguments identified by special
+parameters such as ``static_argnums`` for :func:`jax.jit` or
+``static_broadcasted_argnums`` for :func:`jax.pmap`. Typically, computations
+that involve at least a tracer value will produce a tracer value. Besides tracer
+values, there are **regular** Python values: values that are computed outside JAX
+transformations, or arise from above-mentioned static arguments of certain JAX
+transformations, or computed solely from other regular Python values.
+These are the values that are used everywhere in absence of JAX transformations.
+
+A tracer value carries an **abstract** value, e.g., ``ShapedArray`` with information
+about the shape and dtype of an array. Some tracers, e.g., those that are
+introduced for arguments of autodiff transformations, carry ``ConcreteArray``
+abstract values that actually include the regular array data, and are used,
+e.g., for resolving conditionals. Tracer values computed from these
+concrete tracers, perhaps in combination with regular values, result
+in **concrete tracers**. A **concrete value** is either a regular value or
+a concrete tracer.
+
+Most often values computed from tracer values are themselves tracer values.
+There are very few exceptions, when a computation can be entirely done
+using the abstract value carried by a tracer, in which case the result
+can be a regular value. For example, getting the shape of a tracer
+with ``ShapedArray`` abstract value. Another example, is when explicitly
+casting a concrete tracer value to a regular type: ``int(x)`` or ``float(x)``.
+Another such situation is for ``bool(x)``, which produces a Python bool when
+concreteness makes it possible. That case is especially salient because
+of how often it arises in control flow.
+
+Here is how the transformations introduce abstract or concrete tracers:
```

Comment:
I think "abstract tracers" really means "tracers with abstract values raised above the concrete level." Maybe it'd be good to define that term precisely?

Reply:
The text above said:

> A tracer value carries an **abstract** value, e.g., ``ShapedArray`` with information
> about the shape and dtype of an array.

I have added 
> We will refer here to such tracers as **abstract tracers**.

So that I can refer to them as "abstract tracers". I have not used the word "raised" because 
I do not think it is necessary here to talk about abstraction levels and why a level is higher than others.